src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -625,6 +629,225 @@ bool QgsMapBoxGlStyleConverter::parseLineLayer( const QVariantMap &jsonLayer, Qg
   return true;
 }
 
+bool QgsMapBoxGlStyleConverter::parseCircleLayer( const QVariantMap &jsonLayer, QgsVectorTileBasicRendererStyle &style, QgsMapBoxGlStyleConversionContext &context )
+{
+  if ( !jsonLayer.contains( QStringLiteral( "paint" ) ) )
+  {
+    context.pushWarning( QObject::tr( "%1: Style has no paint property, skipping" ).arg( context.layerId() ) );
+    return false;
+  }
+
+  const QVariantMap jsonPaint = jsonLayer.value( QStringLiteral( "paint" ) ).toMap();
+  QgsPropertyCollection ddProperties;
+
+  // circle color
+  QColor circleFillColor;
+  if ( jsonPaint.contains( QStringLiteral( "circle-color" ) ) )
+  {
+    const QVariant jsonCircleColor = jsonPaint.value( QStringLiteral( "circle-color" ) );
+    switch ( jsonCircleColor.type() )
+    {
+      case QVariant::Map:
+        ddProperties.setProperty( QgsSymbolLayer::PropertyFillColor, parseInterpolateColorByZoom( jsonCircleColor.toMap(), context, &circleFillColor ) );
+        break;
+
+      case QVariant::List:
+      case QVariant::StringList:
+        ddProperties.setProperty( QgsSymbolLayer::PropertyFillColor, parseValueList( jsonCircleColor.toList(), PropertyType::Color, context, 1, 255, &circleFillColor ) );
+        break;
+
+      case QVariant::String:
+        circleFillColor = parseColor( jsonCircleColor.toString(), context );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "%1: Skipping unsupported circle-color type (%2)" ).arg( context.layerId(), QMetaType::typeName( jsonCircleColor.type() ) ) );
+        break;
+    }
+  }
+  else
+  {
+    // defaults to #000000
+    circleFillColor = QColor( 0, 0, 0 );
+  }
+
+  // circle radius
+  double circleRadius = 5.0;
```

Comment:
If this is the radius then the sizes will need to be multiplied by 2 as QGIS uses diameter for sizes.